lib/util.ex

```diff
@@ -75,24 +70,51 @@ defmodule Util do
 
     case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
       {:ok, resp} ->
-        resp.body
-        |> JSON.decode
-        |> format_decoded_response
+        tourney_map = resp.body
+                      |> JSON.decode
+                      |> get_tourney_map
+
+        total_pages = tourney_map
+                      |> Map.get("pageInfo")
+                      |> Map.get("totalPages")
+
+        cond do
+          total_pages < 1 ->
+            IO.puts(:stderr, "No tournament pages found.")
+            System.stop(1)
+          page_num < total_pages ->
+            # There are more pages. Add the entries from this page to the
+            # tourney list and get the next page.
+            this_page = format_decoded_response(tourney_map)
+            get_tourney_page(tourney_list ++ this_page, auth, page_num + 1)
```

Comment:
[The `++` operator is O(length of first argument)](https://hexdocs.pm/elixir/1.12/Kernel.html#++/2). In practice we'll be doing very few iterations and aren't time constrained at all, so it doesn't _really_ matter, but on principle let's flip the arguments so we're only copying the first page rather than the ever-growing `tourney_list`. We don't care about the order anyway.

If we really cared we could prepend with the `|` operator and flatten everything at the end, but that's even more overkill and not worth the footgun if we ever forget to flatten it.